### ISSUE TITLE
Pkldoc generates broken page anchors

### ISSUE THREAD

**Issue by translatenix** [apple/pkl#395]
Searching the standard library docs, I noticed that I always land at the top of the page, not at the function I search for. For example, searching for `sort` takes me to `https://pkl-lang.org/package-docs/pkl/current/base/Collection.html#sort()`. But `#sort()` doesn't exist, so I land at the top of the page. Inspecting the page source, I found that `id` attributes are erroneously percent-encoded:
```
<div id="sort%28%29" class="anchor">
```

**Comment by bioball** [comment]
Thanks for the error report.

Although, it's not strictly necessary, the percent-encoding there is intentional (we are using `java.net.URLEncoder` underneath the hood to encode links).

We should similarly run links from the generated search index through URLEncoder.

**Comment by bioball** [comment]
Yup, it definitely doesn't need to encode parentheses. It'd certainly be an improvement to not encode them here.
